Marlin/configuration_store.cpp

```diff
@@ -1098,7 +1098,7 @@ void Config_ResetDefault() {
     #endif
     SERIAL_EOL;
     #if ENABLED(DISTINCT_E_FACTORS)
-      for (uint8_t i = 0; i < E_STEPPERS; i++) {
+      for (uint8_t i = 0; i < EXTRUDERS; i++) {
```

Comment:
There is a simple problem... in the case of switching extruder we have only one or to stepper for two or four "pseudo" extruder so if you change to extruder higher E_STEPPERS DISTINCT_E_FACTORS don't work. 
I think, if i made it dependet from the SWITCHING_EXTRUDER parameter it can be OK !?